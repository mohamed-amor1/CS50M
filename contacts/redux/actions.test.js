import * as actions from "./actions";

test("updateUser returns an action", () => {
  expect(actions.updateUser({ name: "test name" })).toStrictEqual({
    type: actions.UPDATE_USER,
    payload: { name: "test name" },
  });
  expect(actions.updateUser({ phone: "1234567890" })).toStrictEqual({
    type: actions.UPDATE_USER,
    payload: { phone: "1234567890" },
  });
});
