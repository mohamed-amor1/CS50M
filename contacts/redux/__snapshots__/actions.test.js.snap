// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginUser returns actions dispatches LOG_IN_FULFILLED  with correct creds 1`] = `
[
  {
    "payload": "thisIsATestToken",
    "type": "LOG_IN_FULFILLED",
  },
]
`;

exports[`loginUser returns actions dispatches LOG_IN_REJECTED  with incorrect creds 1`] = `
[
  {
    "payload": "incorrect creds",
    "type": "LOG_IN_REJECTED",
  },
]
`;

exports[`updateUser returns actions handles empty name 1`] = `
{
  "payload": {
    "name": "",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions handles empty object 1`] = `
{
  "payload": {},
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions returns an action 1`] = `
{
  "payload": {
    "name": "test name",
  },
  "type": "UPDATE_USER",
}
`;
